components/camel-fhir/src/main/java/org/apache/camel/component/fhir/FhirTypeConverter.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.fhir;
+
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.rest.api.MethodOutcome;
+import org.apache.camel.Converter;
+import org.apache.camel.Exchange;
+import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Converter
+public final class FhirTypeConverter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FhirTypeConverter.class.getName());
+
+    private FhirTypeConverter() {
+        // Helper class
+    }
+
+    @Converter
+    public static String resourceToString(IBaseResource iBaseResource, Exchange exchange) {
```

Comment:
Hmm maybe such a type converter is not so idea when you require those stuff to be on the exchange property. Sometimes a type converter is not the best thing to add, they should really ideally just be for basic type conversions. Consider removing this if possible.

Reply:
Ah ok I'll remove it. I liked to use it for logging e.g: .log("Executing ${body}") - but this can be done in an application bean or something